applications/sandbox/systems/explosion_particlesystem.hpp

```diff
@@ -0,0 +1,175 @@
+{
+public:
+    ExplosionParticleSystem(explosionParameters params)
+    {
+        m_looping = params.looping;
+        m_maxLifeTime = params.maxLifeTime;
+        m_startingLifeTime = params.startingLifeTime;
+        m_spawnRate = params.spawnRate;
+        m_particleCount = params.particleCount;
+        m_maxParticles = params.maxParticles;
+        m_startingVelocity = params.startingVel;
+        m_startingSize = params.startingSize;
+        m_sizeOverLifetime = params.sizeOverLifeTime;
+        m_particleMaterial = params.particleMaterial;
+        m_particleModel = params.particleModel;
+        m_explosionModel = params.explosionModel;
+
+        m_decelerationScalar = params.decelerationScalar;
+        m_endColor = params.endColor;
+        m_beginColor = m_particleMaterial.get_param<math::vec4>("color");
+    }
+
+    void setup(ecs::component_handle<rendering::particle_emitter> emitter_handle) const override
+    {
+        auto vertPositions = m_explosionModel.get_mesh().get().second.vertices;
+
+        //emitter positions/origin
+        const auto emitterPosHandle = emitter_handle.entity.get_component_handle<position>();
+        auto scaleOfEmitter = emitter_handle.entity.get_component_handle<scale>().read();
+        const math::vec3 emitterPos = emitterPosHandle.read();
+
+        auto emitter = emitter_handle.read();
+
+        for (math::vec3 vert_position : vertPositions)
+        {
+#pragma region Create particle
+            //Checks the emitter if it has a recycled particle to use, if not it creates a new one.
+            ecs::component_handle<rendering::particle> particleComponent = checkToRecycle(emitter_handle);
+
+            auto ent = particleComponent.entity;
+            //Checks if the entity has a transform, if not it adds one.
+            if (!ent.has_components<transform>())
+                ent.add_components<transform>();
+
+            //Gets position, rotation and scale of entity.
+            auto trans = ent.get_component_handles<transform>();
+            auto& [pos, _, scale] = trans;
+
+            //Sets the particle scale to the right scale.
+            auto newPos = emitterPos;
+            pos.write(newPos);
+            scale.write(math::vec3(m_startingSize)* scaleOfEmitter);
+
+            //Populates the particle with the appropriate stuffs.
+            createParticle(particleComponent, trans);
+#pragma endregion 
+#pragma region Set directions
+            math::vec3 pointPos = emitterPos + vert_position;
+            math::vec3 pointDirection = math::normalize(pointPos - emitterPos);
+#pragma endregion
+#pragma region Set parameter values
+            rendering::particle particularParticle = particleComponent.read();
+            particularParticle.lifeTime = 0;
+            particularParticle.particleVelocity = m_startingVelocity.x * pointDirection * scaleOfEmitter.r;
+            particleComponent.write(particularParticle);
+#pragma endregion
```

Comment:
pragma regions should not replace comments, they are very Visual Studio specific! Also if using them try to fix the indenting